src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,180 @@
+                    // Deletes a task from the task list.
+                    String[] splitString = userInput.split(" ", 2);
+
+                    if (splitString.length > 1 && splitString[1].trim().length() > 0) {
+                        int taskNumber = Integer.parseInt(splitString[1].trim());
+                        this.ui.printMessage(this.taskList.deleteTask(taskNumber));
+                        this.storage.save(this.taskList);
+
+                    } else {
+                        throw new DukeException("☹ OOPS!!! Please state which task number "
+                                + "you want to\n delete.");
+                    }
+
+                } else if (type == Parser.CommandType.FIND) {
+                    // Finds the tasks in the task list that contain the String.
+                    String[] splitString = userInput.split(" ", 2);
+
+                    if (splitString.length > 1 && splitString[1].trim().length() > 0) {
+                        String toSearch = splitString[1].trim();
+                        this.ui.printMessage(this.taskList.find(toSearch));
+
+                    } else {
+                        throw new DukeException("☹ OOPS!!! Please give a search query.");
+                    }
+
+                } else if (type == Parser.CommandType.UNKNOWN) {
+                    // Incorrect user input.
+                    throw new DukeException(
+                            "☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+
+                } else {
+                    // Adds a Task to the task list.
+                    String[] split = userInput.split(" ", 2);
+
+                    if (split.length > 1 && split[1].trim().length() > 0) {
+                        String substring = split[1].trim();
+                        Task newTask;
+                        switch (type) {
+                        case TODO:
+                            newTask = new ToDo(substring, false);
+                            break;
+
+                        case DEADLINE:
+                            String[] nameAndDeadline = substring.split(" /by ");
+
+                            if (nameAndDeadline.length > 1
+                                    && nameAndDeadline[1].trim().length() > 0) {
+                                LocalDateTime deadline = Parser
+                                        .formatDateTime(nameAndDeadline[1]);
+                                newTask = new Deadline(nameAndDeadline[0], deadline,
+                                        false);
+
+                            } else {
+                                throw new DukeException("☹ OOPS!!! Please provide a date or "
+                                            + "time for the deadline.");
+                            }
+                            break;
+
+                        default:
+                            String[] nameAndTime = substring.split(" /at ");
```

Comment:
Great and clear naming of variables! Quite consistent throughout your code :smile: